packages/arb-avm-cpp/data_storage/src/combinedmachinecache.cpp

```diff
@@ -21,11 +21,16 @@ void CombinedMachineCache::checkLastMachine(uint256_t& arb_gas_used) {
         last_machine->machine_state.output.arb_gas_used < arb_gas_used) {
         last_machine = nullptr;
     }
+    if (last_last_machine != nullptr &&
+        last_last_machine->machine_state.output.arb_gas_used < arb_gas_used) {
+        last_last_machine = nullptr;
+    }
 }
 
 void CombinedMachineCache::lastAdd(std::unique_ptr<Machine> machine) {
     std::unique_lock lock(mutex);
 
+    last_last_machine = std::move(last_machine);
```

Comment:
I think we should only do this if `last_machine` is non-empty

Reply:
:+1: